.github/workflows/coverage.yaml

```diff
@@ -14,78 +14,81 @@ jobs:
 
     steps:
       - name: Checkout code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
-      - name: Fetch github branches and detect introduces .py files
+      - name: Debug GitHub context
         run: |
-          py_files=()
-          if [ "${{ github.event.pull_request.head.repo.full_name }}" != "${{ github.repository }}" ]; then
-              # Fetch changes when PR comes from remote repo
-              git fetch origin +refs/heads/${{ github.base_ref }}:refs/remotes/origin/${{ github.base_ref }}
-              git fetch origin +refs/pull/${{ github.event.pull_request.number }}/head:refs/remotes/origin/PR-${{ github.event.pull_request.number }}
-              gitdiff=$(git diff --name-status origin/${{ github.base_ref }}...origin/PR-${{ github.event.pull_request.number }} -- tests/)
-          else
-              # Fetch the base branch and the head branch
-              git fetch origin ${{ github.base_ref }}:refs/remotes/origin/${{ github.base_ref }}
-              git fetch origin ${{ github.head_ref }}:refs/remotes/origin/${{ github.head_ref }}
-
-              gitdiff=$(git diff --name-status origin/${{ github.base_ref }}...origin/${{ github.head_ref }} -- tests/)
-          fi
+          echo "Git reference: ${{ github.ref }}"
+          echo "Git head ref: ${{ github.head_ref }}"
+          echo "Git base ref: ${{ github.base_ref }}"
 
-          echo "git diff:"
-          echo "$gitdiff"
-          paths=$(echo "$gitdiff" | grep -oE '/[^[:space:]]+')
-          while IFS= read -r line; do
-            py_files+=("tests$line")
-          done <<< "$paths"
-          echo "Extracted file paths:"
-          for path in "${py_files[@]}"; do
-            echo "$path"
-          done
+      - name: Get all changed python files in tests/ and changes to coverted-ethereum-tests.txt
+        id: changed-tests
+        uses: tj-actions/changed-files@v45
+        with:
+          # TODO: non-test modules such as __init__.py or spec.py could effect coverage - in this case we should
```

Comment:
no, non test files does not affect coverage. when you pass it to fill it just skipped.

Reply:
Yes, `fill` will skip non-test modules, but a `conftest.py` (that defines fixtures that parametrize tests, for example) could affect coverage. I mean, we need to a bit more careful about which tests are effected by these changes and re-fill accordingly.

Reply:
I see. thats a limitation of this script. it does not detect changes in files that are not tests but affect tests